enhancements/image-references.md

```diff
@@ -0,0 +1,225 @@
+build system that will eventually create a bundle image with them.
+
+#### Story 2
+
+As an author of an operand, I would like to create a bundle locally in order
+to test my operator end-to-end on a cluster of my choosing without having to
+edit the core configuration for the operator.
+
+#### Story 3
+
+As an author of a build system, I would like to operate with tooling that allows
+me to clearly define the source of truth for image digests in order to keep the
+build-system-specific configuration to an absolute minimum.
+
+#### Story 4
+
+As an engineer involved in publishing an optional operator, I would like to
+configure semantically equivalent image pull specifications once, in order to not
+need to configure each build system independently.
+
+### Implementation Details/Notes/Constraints
+
+The core problems that must be solved in the implementation of this proposal have
+already been handled in the workflow used in `oc adm release new`. When implementing
+improvements to the `operator-sdk bundle create` process we will simply need to
+create a shared library for the two tools to use. While the shape of the output
+is slightly different and some of the semantics about how the output should be 
+formatted are dissmilar, the core image reference rewriting is identical and the
+process of building a `FROM scratch` image layer is also identical.
+
+Today, some prior art exists in the OSBS workflow for building operator bundles.
+As we improve the Operator SDK tooling to create a straghtforward process for
+creating bundle images, we must make sure a seamless migration is possible.
+
+### Risks and Mitigations
+
+It will be critical that the design be vetted by all of the concerned parties, from
+operator manifest authors to CI system authors and productized pipeline authors to
+ensure that the UX is appropriate in all cases. Furthermore, the largest risk in the
+implementation here is not prioritizing a clean migration pathway for all current
+users who create bundle images, which would lead to further fragmentation of the
+ecosystem, which is directly opposed to the goal of this enhancement.
+
+## Design Details
+
+The definition of a minimally-viable operator bundle image will be changed to
+ensure that all iamge references in the contained manifests have been resolved 
+to a digest and had the pull specifications rewritten to refer to those digests.
+
+The only acceptable process for creating an operator bundle will be to run the
+`operator-sdk bundle create` CLI, providing the manifests, metadata and image
+sources as input to the creation process.
+
+### Proposed UX
+
+Operator manifest authors write a manifest that refers to images using some
+opaque string [ex](https://github.com/openshift/machine-config-operator/blob/master/install/0000_80_machine-config-operator_04_deployment.yaml#L22),
+and provide an `image-references` file alongside their manifests that declares
```

Comment:
The current [on-disk bundle spec](https://github.com/openshift/enhancements/blob/master/enhancements/olm/operator-bundle.md#example-of-an-operator-bundle-that-uses-operator-registry-format) defines a `metadata` directory where arbitrary metadata can be added. 

Maybe we could extend the current spec with some new fields in the `metadata/annotations.yaml` -- or define a new file -- to handle this mapping.

I would also like to make the point that a "kubectlable" resource NOT intended for cluster application could be considered misleading.

Reply:
I would like to keep the implementation as identically similar to that used in `oc adm release new` as possible. Any explicitly different semantics will contribute to maintenance load on the shared library that will execute this logic. I don't have a strong opinion on this being a new file in the `metadata/` directory but for interoperability if it were `image-references` and co-located with manifests (as it is in OpenShift payloads) that would be simplest.

Reply:
A new file makes sense to me, since `annotations.yaml` contains annotations that must appear on a bundle image as `LABEL`s.